src/main/scala/files_metadata/infrastructure/FilesMetaPostgresRepository.scala

```diff
@@ -589,5 +589,72 @@ class FilesMetaPostgresRepository extends FilesMetaRepository {
     }
   }
 
-  override def deleteFileMeta( ownerUuid: UUID, uuid: UUID ): Unit = ???
+  override def deleteFileMeta( uuid: UUID ): Unit = {
+    val connection: Connection = pool.getConnection()
+    connection.setAutoCommit( false )
+
+    try {
+      val getArchiveStatement = connection.prepareStatement(
+        "SELECT * FROM files WHERE uuid = ?"
+      )
+      getArchiveStatement.setObject( 1, uuid )
+      val resultSet = getArchiveStatement.executeQuery()
+      if (resultSet.next()) {
+        val archive_uuid =
+          resultSet.getObject( "archive_uuid" ).asInstanceOf[UUID]
+        val deleteSharedFilesStatement = connection.prepareStatement(
+          "DELETE FROM shared_files WHERE file_uuid = ?"
+        )
+        deleteSharedFilesStatement.setObject( 1, uuid )
+        deleteSharedFilesStatement.executeUpdate()
+        val deleteFileStatement = connection.prepareStatement(
+          "DELETE FROM files WHERE uuid = ?"
+        )
+        deleteFileStatement.setObject( 1, uuid )
+        deleteFileStatement.executeUpdate()
+        val deleteArchiveStatement = connection.prepareStatement(
+          "DELETE FROM archives WHERE  uuid = ?"
+        )
+        deleteArchiveStatement.setObject( 1, archive_uuid )
+        deleteArchiveStatement.executeUpdate()
+
+      }
+      connection.commit()
+    } finally {
+      connection.close()
+    }
+  }
+
+  override def deleteDirectoryMeta( uuid: UUID ): Unit = {
+    val connection: Connection = pool.getConnection()
+    connection.setAutoCommit( false )
+
+    try {
+      val statement = connection.prepareStatement(
+        """
+                SELECT uuid , archive_uuid
+                FROM files_view WHERE
+                parent_uuid = ?
+                """.stripMargin
+      )
+      statement.setObject( 1, uuid )
+
+      val result = statement.executeQuery()
+
+      // Delete each file one by one
+      while (result.next()) {
```

Comment:
This can lead to performance issues. Don't the child files get deleted automatically because of the `ON DELETE CASCADE` in the `files` table?